lib/LaTeXML/Package/beamer.cls.ltxml

```diff
@@ -0,0 +1,1323 @@
+DefMacro('\setjobnamebeamerversion', sub { beamerTODO('setjobnamebeamerversion')});
+
+
+# Beamer has five modes: beamer, second, handout, trans, article.
+# The beamer@mode macro stores the current mode.
+# TODO: Let the user set the actual mode!
+DefMacro('\beamer@mode', 'beamer');
+
+sub getCurrentMode {
+  return ToString(LookupDefinition(T_CS('\beamer@mode'))->getExpansion); }
+
+# Beamer mode specifications (see Overlay Specification below) can additionally specificy two special modes:
+# 'all' (any mode) and 'presentation' (anything except)
+# It can also take the 'all' argument (stuff visible in any mode) or 'presentation' (anything except article).
+
+# matchesMode checks if the mode $needle is contained in $haystack.
+# $needle must be one of the five existing modes.
+sub matchesMode {
+    my ($haystack, $needle) = @_;
+    return 1 if $needle eq $haystack;
+    return 1 if $haystack eq 'all';
+    return 1 if $haystack eq 'handout' && !($needle eq 'article');
+    return 0; }
+
+# like matchesMode, but uses the currentMode as $needle.
+sub matchesCurrentMode {
+  my ($haystack) = @_;
+  matchesMode(ToString($haystack), getCurrentMode()); }
+
+# The '\mode' command exists in three variants:
+# \mode*                   => \beamer@modeoutsideframe
+# \mode<modespec>{stuff}   => \beamer@modeinline<modespec>{stuff}
+# \mode<modespec>          => \beamer@switchmode<modespec>
+DefMacro('\mode', '\@ifstar{\beamer@modeoutsideframe}{\beamer@mode@}');
+DefMacro('\beamer@mode@ Angled', '\@ifnextchar\{{\beamer@modeinline}{\beamer@switchmode}<#1>');
+
+# \mode*
+# ignores all text outside of frames
+# TODO: We don't support this at the moment, perhaps make it a postprocessing option?
+DefMacro('\beamer@modeoutsideframe', '');
+
+# \mode<modespec>{stuff}
+# only do stuff if we are in the provided mode.
+DefMacro('\beamer@modeinline Angled {}', sub {
+  my ($gullet, $spec, $body) = @_;
+  if(matchesCurrentMode(ToString($spec))) {
+    $body->revert; }
+  else {
+    (); } });
+
+# \mode<modespec>
+# if modespec matches, do nothing.
+# else gobble everything until the next \mode (or related command).
+DefMacro('\beamer@switchmode Angled', sub {
+  my ($gullet, $spec) = @_;
+  if(matchesCurrentMode(ToString($spec))) {
+    (); }
+  else {
+    (T_CS('\beamer@processline')); } });
+
+# stop tokens adapted from beamerbasemodes.sty
+my @BEAMER_PROCESSLINE_MARKS = (
+  '\mode', '\mode*',
+  '\article', '\presentation', '\common',
+  '\end{document}', '\begin{document}',
+);
+# skip everything until that token
+DefConstructorI(T_CS('\beamer@processline'), undef, '', reversion => '',
+  afterDigest => [sub {
+      my ($stomach, $whatsit) = @_;
+      my $gullet = $stomach->getGullet;
+      $gullet->readRawLine;
+      my ($unused, $line) = readRawUntilMatch($gullet, @BEAMER_PROCESSLINE_MARKS);
+      # put the stop token back (if we saw it!)
+      $gullet->unread(T_CS($line)) if defined($line); }]);
```

Comment:
This needs a `return;` (or return something/nothing)

Reply:
Done in a80f79cadd77aa9a70d1aefda389c8ce6ab90d40.